### DISCUSSION TITLE
单机Docker部署fe与be时, 提示 backend ip saved in master does not equal to backend local ip

### DISCUSSION THREAD

**Discussion by luoshieryi** [apache/doris#16239] [A - General / Q&A] [answered]
DorisVersion: 1.1.5
DockerVersion: 20.10.17
Dockerfile and shell: https://github.com/apache/doris/tree/master/docker/runtime
Platform: windows11, wsl2(ubuntu)
WLAN局域网IP: 192.168.141.13; WSL IP: 172.20.160.1

在依照文档[构建docker image](https://doris.apache.org/zh-CN/docs/dev/install/construct-docker/construct-docker-image)build 镜像后, 在Windows Powersherll中使用以下指令分别运行fe与be的容器
```
docker run -p 8030:8030 -p 9010:9010 -p 9020:9020 -p 9030:9030 doris-fe-1.1.5 --fe_id 1 --fe_servers fe1:192.168.141.13:9010
```
```
docker run -p 9060:9060 -p 8040:8040 -p 9050:9050 -p 8060:8060 doris-be-1.1.5 --fe_servers fe1:192.168.141.13:9010 --be_addr 192.168.141.13:9050
```

输出日志如下:
```
DEBUG >>>>>> Append the configuration [priority_networks = 192.168.141.0/24] to /opt/apache-doris/be/conf/fe.conf
...
I0130 06:06:03.037909   394 backend_options.cpp:88] priority cidrs in conf: 192.168.141.0/24
I0130 06:06:03.038050   394 backend_options.cpp:73] fail to find one valid non-loopback address, use loopback address.
I0130 06:06:03.038070   394 backend_options.cpp:76] local host ip=127.0.0.1
...
W0130 06:06:08.694612   797 heartbeat_server.cpp:83] backend ip saved in master does not equal to backend local ip192.168.141.13 vs. 127.0.0.1
```

**Comment by FreeOnePlus** [comment, marked as answer]
you can save and run this compose file~
because I haven't submitted the new init script and EntryPoint script in time~
sorry~

-------------- docker-compose.yml -----------------
version: '3'
services:
  docker-fe:
    image: "freeoneplus/doris:1.2.1-fe-x86_64"
    container_name: "doris-fe"
    hostname: "fe"
    environment:
      - FE_SERVERS=fe1:172.20.80.2:9010
      - FE_ID=1
    ports:
      - 8030:8030
      - 9030:9030
    volumes:
      - /data/fe/doris-meta:/opt/apache-doris/fe/doris-meta
      - /data/fe/conf:/opt/apache-doris/fe/conf
    networks:
      doris_net:
        ipv4_address: 172.20.80.2
  docker-be:
    image: "freeoneplus/doris:1.2.1-be-x86_64"
    container_name: "doris-be"
    hostname: "be"
    depends_on:
      - docker-fe
    environment:
      - FE_SERVERS=fe1:172.20.80.2:9010
      - BE_ADDR=172.20.80.3:9050
    ports:
      - 8040:8040
    volumes:
      - /data/be/storage:/opt/apache-doris/be/storage
      - /data/be/conf:/opt/apache-doris/be/conf
      - /data/be/script:/docker-entrypoint-initdb.d
    networks:
      doris_net:
        ipv4_address: 172.20.80.3
networks:
  doris_net:
    ipam:
      config:
        - subnet: 172.20.80.0/16

**Reply by luoshieryi** [reply]
Thank You~ It works for me
By the way, is this blog also written by you? These docker-compose.yml files are similar in some interesting places ( networks-subnet ... )
This blog is very helpful for me~ Thank You
[如何进行 Apache Doris 集群 Docker 快速部署](https://xie.infoq.cn/article/795b598872fc8c5ee2b9df791)
